src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java

```diff
@@ -329,16 +339,18 @@ private void currentBuildResultFailure(AbstractBuild<?, ?> build, BuildListener
             
             //Get the issue-id which was filed when the previous built failed
             String issueId = getIssue(filename);
-            if (issueId != null) {
-                try {
-                    //The status of the issue which was filed when the previous build failed
-                    Status status = getStatus(build, issueId);
+            Status status = getStatus(build, issueId);
```

Comment:
This logic here is attempting to retrieve the JIRA IssueID for the previous build, and then comparing it to a bunch of strings.  I think we want to short-circuit a lot more of this logic -- we want a higher-level `if (forceNewIssue) {` statement that makes sure we never even call `getIssue()`.  Ideally, if `forceNewIssue` is true, not a single statement that would look at a previous build even executes.